software/py/nbf.py

```diff
@@ -99,7 +99,7 @@ def read_objcopy(self, section, output_file):
       print("install riscv-tools first...")
       sys.exit()
 
-    cmd = [objcopy_path, "-O", "verilog", "-j", section, self.riscv_file, output_file]
+    cmd = [objcopy_path, "-O", "verilog", "-j", section, "--set-section-flags", "*bss*=alloc,load,contents", self.riscv_file, output_file]
```

Comment:
What's the purpose of this? Is it supposed to be part of this PR?

Reply:
Yep, very much! This is one of the hardest debugs I've done. Apparently, objcopy doesn't output contents of bss sections and as result, tests were failing because of some failure in the C library. Ideally, elf loader should be able to leave space for bss by looking at load addresses, but the spmd_loader doesn't do that. So, this flag is required.

Reply:
Well, glad you debugged it :)

But is it caused by this PR, or is it a separate bug?

Reply:
We can say, this bug got surfaced in this pr.